File: tracker/src/test/java/org/matomo/sdk/tools/PropertySourceTest.java
package org.matomo.sdk.tools;

import org.junit.Test;

import testhelpers.BaseTest;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.spy;
import static org.mockito.Mockito.verify;


public class PropertySourceTest extends BaseTest {
    @Test
    public void testGetHttpAgent() {
        PropertySource propertySource = spy(new PropertySource());
        propertySource.getHttpAgent();
        verify(propertySource).getSystemProperty("http.agent");

        assertEquals(propertySource.getHttpAgent(), propertySource.getSystemProperty("http.agent"));
    }

    @Test
    public void testGetJVMVersion() {
        PropertySource propertySource = spy(new PropertySource());
        propertySource.getJVMVersion();
        verify(propertySource).getSystemProperty("java.vm.version");

        assertEquals(propertySource.getJVMVersion(), propertySource.getSystemProperty("java.vm.version"));
    }
}

File: tracker/src/test/java/org/matomo/sdk/tools/PropertySourceTest.kt
package org.matomo.sdk.tools

import org.junit.Assert
import org.junit.Test
import org.mockito.Mockito
import testhelpers.BaseTest

class PropertySourceTest : BaseTest() {
    @Test
    fun testGetHttpAgent() {
        val propertySource = Mockito.spy(PropertySource())
        propertySource.httpAgent
        Mockito.verify(propertySource).getSystemProperty("http.agent")

        Assert.assertEquals(propertySource.httpAgent, propertySource.getSystemProperty("http.agent"))
    }

    @Test
    fun testGetJVMVersion() {
        val propertySource = Mockito.spy(PropertySource())
        propertySource.jvmVersion
        Mockito.verify(propertySource).getSystemProperty("java.vm.version")

        Assert.assertEquals(propertySource.jvmVersion, propertySource.getSystemProperty("java.vm.version"))
    }
}
